### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by arj03**
The tests are in general not in a super great shape and they seems rather flaky. This does not fix all of it, but improves a few. I also commented out a few of the noisy console.logs.

This PR fixes so that githubs CI is running. I had to disable a test in async-server-close, otherwise the tests would not run properly. This also happened locally. I think the old way of running tests (that does not work on windows) was masking this problem.

### CODE REVIEW

**Comment by arj03** [comment]
All green :sunglasses:

File: package.json [hunk 1]
```diff
@@ -21,10 +21,11 @@
   "devDependencies": {
     "chloride": "^2.2.8",
     "pull-pushable": "^2.2.0",
+    "tap-spec": "^5.0.0",
     "tape": "^5.0.1"
   },
   "scripts": {
-    "test": "set -e; for t in test/*.js; do node $t; done"
+    "test": "tape test/*.js | tap-spec"
```
Comment (christianbundy) [inline package.json:28]: :spiral_notepad: Note: I think this should be `tape 'test/*.js'` if we want it to run tests on Windows.

**Review by christianbundy — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (arj03) [reply package.json:28]: They do run on Windows :-)

**Comment by arj03** [comment]
Thanks for the quick merge @christianbundy :-)

**Comment by christianbundy** [comment]
Thanks for fixing these tests! Love it.

**Pull request merged**